pkg/providers/tinkerbell/hardware/catalogue_validations.go

```diff
@@ -0,0 +1,214 @@
+package hardware
```

Comment:
The code in this file was copied, verbatim, from `catalogue.go` to isolate it for clarity in `catalogue.go` as it became a little messy. 

In a subsequent PR this will be refactored as it shouldn't be associated with the `Catalogue` construct.